### PR TITLE
[NEW] Wechat OAuth

### LINKED ISSUE #7576: Add WeChat Oauth
Could we have a WeChat login, so we could use it in China,thanks a lot

### PR DESCRIPTION
**Pull request by lhr0909**
Closes #11856 
Closes #7576 

Docs: TODO

This is a newer PR so I can have permissions, merged with #11856 already and https://github.com/xanthous-tech/Rocket.Chat/pull/1

### CODE REVIEW

File: packages/rocketchat-wechat/common.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/* global CustomOAuth */
+import _ from 'underscore';
+
+const config = {
+	serverURL: 'https://open.weixin.qq.com',
```
Comment (Hudell) [inline packages/rocketchat-wechat/common.js]: Are those setting values valid for all users of WeChat?

Comment (lhr0909) [reply packages/rocketchat-wechat/common.js]: @Hudell yes, the ones I am putting in here is valid for all users. The variables are appid, appsecret, and the name fields.

**Comment by lhr0909** [comment]
@Hudell As I am smoothing out our code, we found that our process is already tied into the custom oauth flow, for me it is not ideal, but @c19 and I decided to do so at the time to make it work. I am wondering if we should extend the oauth class and override the methods to make this wechat flow work. Please advice.

**Comment by Hudell** [comment]
Were you able to get things working? Can I help with anything?

**Comment by lhr0909** [comment]
> Were you able to get things working? Can I help with anything?

@Hudell Hi, we haven't had enough time to extract the wechat related into its own class cleanly. I think we need to override most methods. Any class we can follow off of?

**Comment by rodrigok** [comment]
Closing this PR since it has no extra commits based on #11856 and it's behind it

**Pull request closed without merging**